Source/Light.swift

```diff
@@ -14,19 +14,20 @@ public struct Light: Equatable, CustomStringConvertible {
 	public let connected: Bool
 	public let group: Group?
 	public let location: Location?
+    public let touchedAt: NSDate?
 
 	public func toSelector() -> LightTargetSelector {
 		return LightTargetSelector(type: .ID, value: id)
 	}
 
 	func lightWithProperties(power: Bool? = nil, brightness: Double? = nil, color: Color? = nil, connected: Bool? = nil) -> Light {
-		return Light(id: id, power: power ?? self.power, brightness: brightness ?? self.brightness, color: color ?? self.color, label: label, connected: connected ?? self.connected, group: group, location: location)
+		return Light(id: id, power: power ?? self.power, brightness: brightness ?? self.brightness, color: color ?? self.color, label: label, connected: connected ?? self.connected, group: group, location: location, touchedAt: NSDate())
```

Comment:
This is why you're getting fresh dates. Might be better to make it an optional argument and default it to the current date. That way we're making less assumptions about how it's being used and it's more consistent with the other properties.